### PR TITLE
First stab at documenting the exec.d support added by #29

### PR DESCRIPTION
**Pull request by chuckbutler**
Mostly a copy/paste operation with some fluff to drive the idea home
that this is for custom support of pre-conditions.

Addresses #54

### CODE REVIEW

**Comment by chuckbutler** [comment]
@johnsca will you review my words? It's just words... they may not even be the best words.

**Comment by marcoceppi** [comment]
These words are huuuggeeeeee :tangerine:. LGTM :+1:

File: README.md [hunk 1]
```diff
@@ -206,6 +206,22 @@ def my_opt_changed():
...
 This layer currently does not define any actions.
 
+# Exec.d Support
+
+This is a mechanism used to customize provisioned machines so the charms can
+actually work. This is useful if there are pre-conditions in your cloud that
+you must account for, such as charm pre-requisits (pre-reactive install) or
+environment customization.
+
+The feature implementation documented this as the following:
```
Comment (johnsca) [inline README.md]: Maybe call this an example use-case instead?  I'm not clear on what "the feature implementation documented this as" means.

File: README.md [hunk 2]
```diff
@@ -206,6 +206,22 @@ def my_opt_changed():
 
 This layer currently does not define any actions.
 
+# Exec.d Support
+
+This is a mechanism used to customize provisioned machines so the charms can
+actually work. This is useful if there are pre-conditions in your cloud that
+you must account for, such as charm pre-requisits (pre-reactive install) or
```
Comment (johnsca) [inline README.md]: "prerequisites"

**Comment by johnsca** [comment]
Merging this as it has sat for too long, and documentation with minor nits is better than no documentation.

**Pull request merged**